go-controller/pkg/clustermanager/pod/allocator.go

```diff
@@ -209,6 +210,15 @@ func (a *PodAllocator) releasePodOnNAD(pod *corev1.Pod, nad string, podDeleted,
 
 	hasIPAM := util.DoesNetworkRequireIPAM(a.netInfo)
 	hasIDAllocation := util.DoesNetworkRequireTunnelIDs(a.netInfo)
+	hasPersistentIPs := network.IPAMClaimReference != ""
+	if hasPersistentIPs {
+		ipamClaim, err := a.findIPAMClaim(network)
+		if err != nil {
+			hasPersistentIPs = false
+		} else {
+			hasPersistentIPs = hasPersistentIPs && ipamClaim != nil && len(ipamClaim.Status.IPs) > 0
```

Comment:
hasPersistentIPs will be always true here since we have the if statement, so we don't need to add it I think this is enough 
```golang
hasPersistentIPs = ipamClaim != nil && len(ipamClaim.Status.IPs) > 0
```